.github/workflows/ci.yml

```diff
@@ -496,9 +496,13 @@ jobs:
           sudo apt-get -y update
 
       - name: Install TiUP
-        run: |
-          curl --proto '=https' --tlsv1.2 -sSf https://tiup-mirrors.pingcap.com/install.sh | sh
-          ~/.tiup/bin/tiup -v
+        uses: nick-fields/retry@v2
+        with:
+          timeout_minutes: 10
+          max_attempts: 5
+          command: |
+            curl --proto '=https' --tlsv1.2 -sSf https://tiup-mirrors.pingcap.com/install.sh | sh
+            ~/.tiup/bin/tiup -v
```

Comment:
we could avoid introducing an external GH Action with this:
```
     - name: Install TiUP
        timeout-minutes: 10
        run: |
          while ! ~/.tiup/bin/tiup -v; do
            curl --proto '=https' --tlsv1.2 -sSf https://tiup-mirrors.pingcap.com/install.sh | sh
          done
```

Reply:
Awesome! I like that. Let me use that instead. Thanks!

Reply:
Done. Thanks @sgirones!